hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -442,14 +443,19 @@ private ResponseCodeEnum validateFeeSchedule(List<CustomFeesOuterClass.CustomFee
 					}
 				}
 			} else if (customFee.hasFractionalFee()) {
-
 				/* TODO: Should fee collectors for fractional fees automatically be associated to the newly
 				    created token?  (This will require their keys to sign the TokenCreate transaction.) */
 				final var fractionalSpec = customFee.getFractionalFee();
-				final var fraction = fractionalSpec.getFractionOfUnitsToCollect();
+				final var fraction = fractionalSpec.getFractionalAmount();
 				if (fraction.getDenominator() == 0) {
 					return FRACTION_DIVIDES_BY_ZERO;
 				}
+				if (!signsMatch(fraction.getNumerator(), fraction.getDenominator())) {
+					return CUSTOM_FEE_MUST_BE_POSITIVE;
+				}
+				if (fractionalSpec.getMaximumAmount() < 0 || fractionalSpec.getMinimumAmount() < 0) {
+					return CUSTOM_FEE_MUST_BE_POSITIVE;
```

Comment:
Should there be a check that MaximumAmount > MinimumAmount ?

Reply:
Also done!!